snap/container.go

```diff
@@ -44,6 +44,9 @@ type Container interface {
 	// Install copies the snap file to targetPath (and possibly unpacks it to mountDir)
 	Install(targetPath, mountDir string) error
 
+	// PreRemove gets a snap ready for removal
+	PreRemove() error
```

Comment:
I started doing that but found it doesn't really help things (and instead duplicates a bit of code), as we still need the rm -rf for try leftovers. If you think it's still worth it, I'll do that.